src/records.rs

```diff
@@ -146,7 +145,20 @@ impl Records {
             asset_records: Persistable::new(),
             nullifier_records: Persistable::new(),
         };
-        records.reload();
+        for record in records.store.iter() {
+            let ro = &record.ro;
+            records.asset_records.insert((
+                (
+                    ro.asset_def.code,
+                    ro.pub_key.address().clone(),
+                    ro.freeze_flag,
+                ),
+                (record.ro.amount.into(), record.uid),
+            ));
+            records
+                .nullifier_records
+                .insert((record.nullifier, record.uid));
+        }
```

Comment:
This still has the issue of returning from `new` with some `pending_changes` buffered in `asset_records` and `nullifier_records`. We may need to do `pending_changes.commit(); asset_records.commit()` before returning so that we return without any pending changes (note: not `self.commit()`, because committing the underlying store without being explicitly asked to is a no-no)

Reply:
Oh yeah that's right, I think commiting `asset_records` and `nullifier_records` should be correct.